Userland/Applications/PixelPaint/PaletteWidget.cpp

```diff
@@ -73,56 +79,24 @@ PaletteWidget::PaletteWidget()
     top_color_container.layout()->set_spacing(1);
 
-    auto& bottom_color_container = color_container.add<GUI::Widget>();
+    auto& bottom_color_container = m_color_container->add<GUI::Widget>();
+    bottom_color_container.set_name("bottom_color_container");
     bottom_color_container.set_layout<GUI::HorizontalBoxLayout>();
     bottom_color_container.layout()->set_spacing(1);
 
-    auto add_color_widget = [&](GUI::Widget& container, Color color) {
-        auto& color_widget = container.add<ColorWidget>(color, *this);
-        color_widget.set_fill_with_background_color(true);
-        auto pal = color_widget.palette();
-        pal.set_color(ColorRole::Background, color);
-        color_widget.set_palette(pal);
-    };
-
-    add_color_widget(top_color_container, Color::from_rgb(0x000000));
-    add_color_widget(top_color_container, Color::from_rgb(0x808080));
-    add_color_widget(top_color_container, Color::from_rgb(0x800000));
-    add_color_widget(top_color_container, Color::from_rgb(0x808000));
-    add_color_widget(top_color_container, Color::from_rgb(0x008000));
-    add_color_widget(top_color_container, Color::from_rgb(0x008080));
-    add_color_widget(top_color_container, Color::from_rgb(0x000080));
-    add_color_widget(top_color_container, Color::from_rgb(0x800080));
-    add_color_widget(top_color_container, Color::from_rgb(0x808040));
-    add_color_widget(top_color_container, Color::from_rgb(0x004040));
-    add_color_widget(top_color_container, Color::from_rgb(0x0080ff));
-    add_color_widget(top_color_container, Color::from_rgb(0x004080));
-    add_color_widget(top_color_container, Color::from_rgb(0x8000ff));
-    add_color_widget(top_color_container, Color::from_rgb(0x804000));
-
-    add_color_widget(bottom_color_container, Color::from_rgb(0xffffff));
-    add_color_widget(bottom_color_container, Color::from_rgb(0xc0c0c0));
-    add_color_widget(bottom_color_container, Color::from_rgb(0xff0000));
-    add_color_widget(bottom_color_container, Color::from_rgb(0xffff00));
-    add_color_widget(bottom_color_container, Color::from_rgb(0x00ff00));
-    add_color_widget(bottom_color_container, Color::from_rgb(0x00ffff));
-    add_color_widget(bottom_color_container, Color::from_rgb(0x0000ff));
-    add_color_widget(bottom_color_container, Color::from_rgb(0xff00ff));
-    add_color_widget(bottom_color_container, Color::from_rgb(0xffff80));
-    add_color_widget(bottom_color_container, Color::from_rgb(0x00ff80));
-    add_color_widget(bottom_color_container, Color::from_rgb(0x80ffff));
-    add_color_widget(bottom_color_container, Color::from_rgb(0x8080ff));
-    add_color_widget(bottom_color_container, Color::from_rgb(0xff0080));
-    add_color_widget(bottom_color_container, Color::from_rgb(0xff8040));
+    auto result = load_palette_file("/res/color-palettes/default.palette");
+    if (result.is_error())
+        GUI::MessageBox::show_error(window(), String::formatted("Loading default palette failed: {}", result.error()));
```

Comment:
It would be nice if `load_palette_file` returned a `Result<Vector<Color>, String>` and not do any side-effects, and then you called `populate_color_container` here manually on the result. This way, in case we want later to e.g. preview palettes before selecting them, we can just re-use the `load_palette_file` call directly :)

Reply:
Good point

Reply:
It's static now and (I think) free from side effects :)

Reply:
Great :) Feel free to directly mark all of these conversations as "resolved" when you've addressed them. This way, other reviewers know that they've been dealt with!